src/Augmint.ts

```diff
@@ -205,13 +200,128 @@ export class Augmint {
+        const token: AugmintToken = new AugmintToken(tokenContract.connect(this.web3), this.ethereumConnection);
+        return loanManager.repayLoan(loan, repaymentAmount, userAccount, token);
+    }
+
+    public async newEthBackedLoan(loanProduct: LoanProduct, weiAmount: Wei, userAccount: string, minRate?:Tokens): Promise<Transaction> {
+        const loanManager: LoanManager = this.getLoanManagerByAddress(loanProduct.loanManagerAddress)
+        return loanManager.newEthBackedLoan(loanProduct, weiAmount, userAccount, minRate);
+    }
+
+    public async getAllLoans(): Promise<Loan[]> {
+        const result: Loan[] = [];
+        for (const loanManager of this.getAllLoanManagers()) {
+            const lmLoans: Loan[] = await loanManager.getAllLoans();
+            result.push(...lmLoans);
+        }
+        return result;
+    }
+
+    public async getLoansToCollect(): Promise<Loan[]> {
+        try {
+            const allLoans: Loan[] = await this.getAllLoans();
+            return allLoans.filter((loan: Loan) => loan.isCollectable);
+        } catch (error) {
+            throw new Error("fetchLoansToCollectTx failed.\n" + error);
+        }
+    }
+
+    public collectLoans(loansToCollect: Loan[], userAccount: string): Transaction[] {
+        let currentAddress: string | null = null;
+        let loanManager: LoanManager | null = null;
+        const result: Transaction[] = [];
+        let loans: Loan[] = [];
+        loansToCollect.sort((a: Loan, b: Loan) => (a.loanManagerAddress < b.loanManagerAddress ? 1 : -1));
+        for (const loan of loansToCollect) {
+            if (currentAddress !== loan.loanManagerAddress) {
+                if (currentAddress !== null && loanManager !== null) {
+                    result.push(loanManager.collectLoans(loans, userAccount));
+                }
+                loanManager = this.getLoanManagerByAddress(loan.loanManagerAddress);
+                loans = [];
+            }
+            currentAddress = loan.loanManagerAddress;
+            loans.push(loan);
+        }
+        if (loanManager) {
+            result.push(loanManager.collectLoans(loans, userAccount));
+        }
+        return result
+    }
+
+    public async getLoanCounts(): Promise<ILoanCount[]> {
+        const result:ILoanCount[] = [];
+        for (const loanManager of this.getAllLoanManagers()) {
+            const loanCount:number = await loanManager.getLoanCount();
+            result.push({
+                loanManagerAddress: loanManager.address,
+                loanCount
+            })
+        }
+        return result;
+    }
+
+    public addExtraCollateral(loan: Loan, weiAmount: Wei, userAccount: string ): Transaction | undefined {
+        const loanManager: LoanManager = this.getLoanManagerByAddress(loan.loanManagerAddress)
+        return loanManager.addExtraCollateral(loan, weiAmount, userAccount)
+    }
+
+    private getLoanManager(contract: DeployedContract<LoanManagerInstance>): LoanManager {
```

Comment:
an eager initialization in the constructor would be a lot cleaner. there's no resources saved by lazy init, since we'll be using all loanmanagers anyway.

Reply:
done

Reply:
`getLoanManagerByAddress` and `getAllLoanManagers` should use this map directly then.

Reply:
done